pkg/reconciler/metric/controller.go

```diff
@@ -66,5 +66,7 @@ func NewController(
 		},
 	})
 
+	collector.Watch(impl.EnqueueKey)
```

Comment:
Here it presumes metrics name == pa name.
Probably fine, but 🤷‍♂

Reply:
I don't think it does? The collector manages metrics and this keys off of metrics. Both are Metric names.